test/integration/kubernetes_deploy_test.rb

```diff
@@ -632,6 +632,18 @@ def test_can_deploy_deployment_with_zero_replicas
     ])
   end
 
+  def test_can_deploy_deployment_with_partial_rollout_success
+    result = deploy_fixtures("slow-cloud", subset: ["configmap-data.yml", "web.yml.erb"])
+    assert_deploy_success(result)
+
+    result = deploy_fixtures("slow-cloud", subset: ["configmap-data.yml", "web.yml.erb"])
+    assert_deploy_success(result)
+
+    assert_logs_match_all(
+      [%r{Deployment\/web\s+[34] replicas, 3 updatedReplicas, 2 availableReplicas, [12] unavailableReplica}]
```

Comment:
could `maxSurge: 0` remove the variability here?

Reply:
I thought it might, but after trying it doesn't, just changes the combinations.